seldon-core-operator/tox.ini

```diff
@@ -19,17 +19,18 @@ allowlist_externals =
     tox
     rockcraft
 deps =
+    charmed_kubeflow_chisme
     juju~=2.9.0
     pytest
     pytest-operator
     ops
 commands =
     # build and pack rock
     rockcraft pack
-    bash -c 'NAME=$(yq eval .name rockcraft.yaml) && \
-             VERSION=$(yq eval .version rockcraft.yaml) && \
-             ARCH=$(yq eval ".platforms | keys" rockcraft.yaml | awk -F " " '\''{ print $2 }'\'') && \
-             ROCK="$\{NAME\}_$\{VERSION\}_$\{ARCH\}" && \
+    bash -c 'ROCK=$(python -c '\''from charmed_kubeflow_chisme.rock import CheckRock; \
```

Comment:
I think this is a little bit overkill, considering that you can achieve the same with yq, a much lighter package than chisme.

Reply:
Yes, that was first iteration. Then test class was created that is used in ROCK tests. It fetches exactly same info as 6 lines of yq code, but is also used in python code.

Reply:
As you can see in the diff, `yq` CLI commands wer the original impementation. It is absolute nightmare to debug and needs to be copied in multiple places in rockcraft.yaml.
The solution with special python class executed as a script is shorter, easy to maintain. If, let's say, we need to change the name of the rock to conform to some other naming format we would need to update all rockcraft.yaml files with new yq commands.
In addition, the same Python class is used in rock tests.

Reply:
> As you can see in the diff, yq CLI commands were the original impementation. 

I'm not questioning that, I can see that yq was the tool for this, I think we should keep it.

> It is absolute nightmare to debug and needs to be copied in multiple places in rockcraft.yaml.

Why do you need to copy this in multiple places in rockcraft.yaml? This piece of script is used for extracting data from `rockcraft.yaml`, which is stored in vars and then passed to other commands.

> The solution with special python class executed as a script is shorter, easy to maintain. If, let's say, we need to change the name of the rock to conform to some other naming format we would need to update all rockcraft.yaml files with new yq commands.

I don't think you need to update any `yq` commands, as you wouldn't update anything in the [chisme implementation](https://github.com/canonical/charmed-kubeflow-chisme/blob/main/src/charmed_kubeflow_chisme/rock/_check_rock.py), since you are extracting data from a yaml file that will always have a `name` and `platforms` key.

Reply:
`yq` was just a prototype. 
Extraction of data from rockcraft.yaml is done in multiple places in `tox.ini` (unit and integration) and there is no way (could not figure out) how to pass that throught variables from one part of `tox.ini` to another.
Regarding `yq` commads maintability. Currently, ROCK is formatted [`{name}_{version}_{arch}`](https://github.com/canonical/charmed-kubeflow-chisme/blob/main/src/charmed_kubeflow_chisme/rock/_check_rock.py#LL26C18-L26C42). If we use `yq` commands in `tox.ini` then if that format changes for some reason (removing `arch` for example) we will need to update all `tox.ini`.
If it is done in Chisme's CheckRock class, we would only need to update it in that class.

Reply:
I understand @i-chvets' point of having the parsing logic in one place, in case the format changes. But tbh I don't think this should/would happen. 

So for now I'm also leaning more towards just keeping `yq` for now rather than invoking Python code and importing a whole package just for some parsing.

Reply:
Argument against importing of the package is its size which is measured in KB. In comparison to other imported packages that are mesured is MB this argument is not convincing.
ROCK naming will most likely change, when adoption increases.

I will revert to useing `yq` commands.
**NOTE:** Reverting to using bash shell commands significantly reduces maintability of `tox.ini` ACK: @kimwnasptd